controllers/front/ajax.php

```diff
@@ -44,28 +49,28 @@ public function postProcess()
     private function submitHostedFields()
     {
         try {
-            if (!Order::getOrderByCartId($this->context->cart->id)) {
-                $this->validateOrder();
+            if (Order::getOrderByCartId($this->context->cart->id)) {
+                $this->ajaxDie(json_encode([
+                    'error' => true,
+                    'message' => $this->module->l('Order already exists', self::FILE_NAME),
+                    'url' => $this->getRedirectionToControllerUrl('fail'),
+                ]));
             }
 
-            /** @var SaferPayCardAliasRepository $cardAliasRep */
-            $cardAliasRep = $this->module->getService(SaferPayCardAliasRepository::class);
-
-            $selectedCard = Tools::getValue('selectedCard');
-
-            $alias = $cardAliasRep->getSavedCardAliasFromId($selectedCard);
-
-            /** @var SaferPayInitialize $initializeService */
-            $initializeService = $this->module->getService(SaferPayInitialize::class);
-            $initializeBody = $initializeService->initialize(
+            // refactor it to create checkout data from validator request
+            $checkoutData = CheckoutData::createFromRequest(
```

Comment:
would more likely be "create" for the name of function, as you are not passing a request here